pkg/quota/evaluator/core/persistent_volume_claims.go

```diff
@@ -30,19 +30,37 @@ import (
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
+const storageClassSuffix string = ".storage-class.kubernetes.io/"
+
 // NewPersistentVolumeClaimEvaluator returns an evaluator that can evaluate persistent volume claims
 func NewPersistentVolumeClaimEvaluator(kubeClient clientset.Interface) quota.Evaluator {
-	allResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	fixedResources := []api.ResourceName{api.ResourcePersistentVolumeClaims, api.ResourceRequestsStorage}
+	usageFunc := PersistentVolumeClaimUsageFunc
+	constraintsFunc := makePersistentVolumeClaimConstraintsFunc(usageFunc)
 	return &generic.GenericEvaluator{
 		Name:              "Evaluator.PersistentVolumeClaim",
 		InternalGroupKind: api.Kind("PersistentVolumeClaim"),
-		InternalOperationResources: map[admission.Operation][]api.ResourceName{
-			admission.Create: allResources,
+		Operations:        []admission.Operation{admission.Create},
+		MatchedResourceNamesFunc: func(items []api.ResourceName) (result []api.ResourceName) {
+			for _, item := range items {
+				if quota.Contains(fixedResources, item) {
+					result = append(result, item)
+					continue
+				}
+				// match pvc resources scoped by storage class (<storage-class-name>.storage-class.kubernetes.io/<resource>)
+				for _, resource := range fixedResources {
```

Comment:
Do we have validation on resource names with lengths?  Length would have to be `MAX_STORAGE_CLASS_NAME + len(RESOURCE_NAME) + 1`